torch/distributed/algorithms/_checkpoint/_checkpoint_wrapper.py

```diff
@@ -17,22 +19,28 @@ def __init__(
         self,
         mod: torch.nn.Module,
         checkpoint_impl: CheckpointImpl = CheckpointImpl.REENTRANT,
+        offload_to_cpu: bool = False,
     ):
         super().__init__()
         self.mod = mod
         self.checkpoint_impl = checkpoint_impl
+        self.offload_to_cpu = offload_to_cpu
 
     def forward(self, *args, **kwargs):
-        return checkpoint(
-            self.mod,
-            use_reentrant=(self.checkpoint_impl == CheckpointImpl.REENTRANT),
-            *args,
-            **kwargs,
-        )
+        offload_mgr = save_on_cpu(pin_memory=True) if self.offload_to_cpu else suppress()
+        with offload_mgr:
```

Comment:
so basically inner activations are not saved during checkpoint(), save_on_cpu() will just save outer activation tensors on cpu?

Reply:
Yeah, inner activations (with the reentrant-based checkpointing impl that's the only one currently supported) will be discarded since we run under torch.no_grad mode. 

All tensors saved via autograd (including those with `save_for_backward` in the checkpointing impl) will be offloaded to CPU as well, and these are the outer activations.